lib/types.js

```diff
@@ -18,14 +19,16 @@ const exclude = function (types, excludes) {
     return types;
   }
 
-  for (const doc of types) {
+  const typesCopy = cloneDeep(types);
```

Comment:
now cloning `types` as needed so we don't mutate the one that `graphql-tag` has cached

Reply:
it's about 50% slower than w/o cloning